src/lib/logical_query_plan/mock_node.cpp

```diff
@@ -118,14 +124,17 @@ size_t MockNode::_on_shallow_hash() const {
 std::shared_ptr<AbstractLQPNode> MockNode::_on_shallow_copy(LQPNodeMapping& node_mapping) const {
   const auto mock_node = MockNode::make(_column_definitions, name);
   mock_node->set_table_statistics(_table_statistics);
+  mock_node->set_soft_key_constraints(_table_key_constraints);
+  mock_node->set_functional_dependencies(_functional_dependencies);
```

Comment:
The constraints will be required for test cases in the upcoming LQP unique constraints PR. However, it is not planned to generate FDs based on table constraints as it is done by StoredTableNodes. MockNodes should only provide container functionality for FDs and table constraints.

Reply:
Shall I remove it anyway?

Reply:
How about we allow only one of the two to be set (to avoid inconsistencies) and prevent the FDs to be retrieved when soft key constraints are set (to avoid the user expecting the FDs to be derived from the constraints)?

Reply:
Hmm, is this still simple enough for MockNode? Since MockNode is for testing purposes only, I could live with pure container functionality. 

I just noticed: In case of MockNode, it does not make sense to use the term "soft" for constraints. There is no data anyway. Nothing to check for.

Reply:
> Hmm, is this still simple enough for MockNode? Since MockNode is for testing purposes only, I could live with pure container functionality.

We wouldn't add any logic, we just add assertions to prevent some constellations that are probably prone to errors.

> I just noticed: In case of MockNode, it does not make sense to use the term "soft" for constraints. There is no data anyway. Nothing to check for.

Good point. I can see arguments for both changing it and keeping it (to mirror the existing calls in a StoredTableNode). Your call.

Reply:
> We wouldn't add any logic, we just add assertions to prevent some constellations that are probably prone to errors.

Good idea. I added one.

> Good point. I can see arguments for both changing it and keeping it (to mirror the existing calls in a StoredTableNode). Your call.

Ok. I decided against "soft" to avoid confusion.